File: leetcode/27. Remove Element.py
def removeElement(nums: [int], val: int) -> int:
    count = 0
    for i in nums:
        if i == val: 
            nums[count]=i
            count+=1
    return count

val = 3
nums = [3,2,2,3]
print(removeElement(nums,val))

File: leetcode/27. Remove Element.cpp
#include<bits/stdc++.h>
using namespace std;

int removeElement(vector<int>& nums, int val){
    int counts = 0;
    for(int i=0; i<nums.size(); i++){
        if(nums[i]!=val) nums[counts++] = nums[i];
    }
    return counts;
}

int main(){
    vector<int> nums = {0,1,2,2,3,0,4,2};
    cout<<removeElement(nums,2);
    for(auto element : nums) cout<<"\t"<<element;
    return 0;
}

// https://leetcode.com/problems/remove-element/